### PR TITLE
MR-757 - Add API request to change Asset Management (required for LBH Ownership edit feature)

### PR DESCRIPTION
**Pull request by ange47rm**
Prior to merging this PR/as of now, `patchAsset` is only set up for editing the `boilerHouseId` of an asset (I have checked the other repos to make sure there are no other MFE referencing this method, other than Common and Property - [Github references](https://github.com/search?q=org%3ALBHackney-IT%20patchasset&type=code))

This PR enables patchAsset to accept a new type of request, to change asset ownership (`isCouncilProperty` field).

Changes:
- Renamed `PatchAssetRequest` interface to `PatchAssetBoilerHouseRequest`, more specific. 
- Added `PatchAssetLbhOwnershipRequest` interface, to be used when editing asset ownership.
- Updated AssetManagement interface to be aligned with its [counterpart class in Asset Shared](https://github.com/LBHackney-IT/asset-shared/blob/main/Hackney.Shared.Asset/Domain/AssetManagement.cs) (v 0.37 - latest). To avoid any potential errors, I've only added the remaining fields (that some asset may or may not have) as optional.
- `patchAsset` now can accepts requests/parameters of type PatchAssetBoilerHouseRequest or PatchAssetLbhOwnershipRequest
- Added test for this new request type (`PatchAssetLbhOwnershipRequest`) and mock request object.

Manually tested with Postman and via the MMH front-end the following:
- add boiler house ID
- remove boiler house ID
- changed `isCouncilProperty` (from development branch of Property - [see related PR](https://github.com/LBHackney-IT/mtfh-frontend-property/pull/110))
- checked that, when changing the `isCouncilProperty` field, only this field of the asset and the asset version are changed, and made sure other fields are not affected/lost.

### CODE REVIEW

File: lib/api/asset/v1/types.ts [hunk 1]
```diff
@@ -61,6 +61,13 @@ export interface AssetManagement {
...
   owner: string;
   isTMOManaged: boolean;
+  propertyOccupiedStatus?: string;
+  propertyOccupiedStatusReason?: string;
+  isNoRepairsMaintenance?: boolean;
+  councilTaxType?: string;
+  councilTaxLiability?: string;
+  isTemporaryAccomodation?: boolean;
+  readyToLetDate?: boolean;
 }
```
Comment (ange47rm) [inline lib/api/asset/v1/types.ts:72]: For reference:
image

File: lib/api/asset/v1/types.ts [hunk 2]
```diff
@@ -138,6 +145,10 @@ export interface GetAssetParentsResponse {
...
 
-export interface PatchAssetRequest {
+export interface PatchAssetBoilerHouseRequest {
   boilerHouseId: string;
 }
+
+export interface PatchAssetLbhOwnershipRequest {
+  assetManagement: AssetManagement;
+}
```
Comment (LBHCallumM) [inline lib/api/asset/v1/types.ts:154]: Nice

File: lib/api/asset/v1/types.ts [hunk 3]
```diff
@@ -61,6 +61,13 @@ export interface AssetManagement {
   managingOrganisationId: string;
   owner: string;
   isTMOManaged: boolean;
+  propertyOccupiedStatus?: string;
+  propertyOccupiedStatusReason?: string;
+  isNoRepairsMaintenance?: boolean;
+  councilTaxType?: string;
+  councilTaxLiability?: string;
+  isTemporaryAccomodation?: boolean;
+  readyToLetDate?: boolean;
```
Comment (LBHCallumM) [inline lib/api/asset/v1/types.ts:70]: Are all of these required for this PR? Could the rest of these be part of a separate PR?

**Review by LBHCallumM — APPROVED** [review]
LGMT. Added a question about adding all of the new fields.

File: lib/api/asset/v1/types.ts (hunk 3 above)
Comment (ange47rm) [reply lib/api/asset/v1/types.ts:70]: They're all required, I did try just sending isCouncilProperty via Postman but the API didn't like it unfortunately
